frontend/src/metabase-lib/v1/parameters/utils/targets.ts

```diff
@@ -133,10 +133,12 @@ export function buildDimensionTarget(
 
 export function buildColumnTarget(
   query: Lib.Query,
-  stageIndex: number,
+  nonNegativeStageIndex: number,
   column: Lib.ColumnMetadata,
   parameter: Parameter | undefined,
 ): StructuredParameterDimensionTarget {
+  // BE works incorrectly with non-negative stage indexes: https://github.com/metabase/metabase/issues/48441
+  const stageIndex = nonNegativeStageIndex - Lib.stageCount(query);
```

Comment:
The double negative might be tripping me up here, but I read the `nonNegativeStageIndex` argument as: only pass a positive stage index here.

But then the comment seems to say that the BE works incorrectly with non-negative stage indexes.

And then the code seems to fix the issue, making it unnecessary for the argument to be non-negative at all.

On another note can we quick-fix this on the FE by doing something like

```suggestion
  const fixedStageIndex = stageIndex > 0 ? stageIndex - Lib.stageCount(query) : stageIndex;
```
That way callers don't need to care about the stage index?

Reply:
68ccdb6dbbd1488203e9bb77efb06ebb0ea263a4